deploy/kubernetes/helm/che/templates/_configMapHelper.tpl

```diff
@@ -0,0 +1,19 @@
+{{- define "properties" -}}
+  {{- range $key, $value := . -}}
+    {{- $key | upper | replace "_" "__" | nindent 2 -}}
```

Comment:
It is also needed to replace '.' on '_' otherwise  `che.port` is translated into `CHE.PORT`

Reply:
Nice catch. I'll try to fix it.

Reply:
@tolusha I weighed the use-case. IMO, It's better to stop use `che.port` as the key.
Instead, we can write like this.

```
che:
  port: "8080"
```

Otherwise, `CHE_PORT` from `che.port: 8080` and `CHE_PORT` from `{"che": { "port": 8080}}` will conflict.

Reply:
It's easy to add the code for replacing to `.`. But I guess users may be confused ...
 
```suggestion
    {{- $key | upper | replace "_" "__" | replace "." "_" | nindent 2 -}}
```

Reply:
Yes, but there is common rule which we translate `che.properties` entries into env variables.
`_` -> `__` and `.` -> `_` , so I would like to use it here also and having the fact that `x.y` is easily to write then `{ x: {y : } }`.
If you don't mind I would commit you suggestion.

Reply:
BTW,  the config map eventually will contain the latest value.

Reply:
> If you don't mind I would commit you suggestion.

Hmm..., I can confirm `x.y` style is convenient for Che core developers who improve new function.
So I agree with your suggestion if we can confirm `x.y` -> `X_Y` is just for a testing.

Reply:
Not sure if I understood you.
What do you mean by "just for a testing" ?

Reply:
I mean "we should not merge any keys named `x.y` for `X_Y` into `values.yaml` on the master branch."
Otherwise, users can't set custom values with `helm --set x.y=value`.
Because Helm will convert `--set x.y=value` to `{x: { y:  value}}`.

Even though, just in case under development / testing phase, I agree `x.y_z` -> `X_Y__Z` will be convenient.